torch_topological/nn/multi_scale_kernel.py

```diff
@@ -0,0 +1,138 @@
+import torch
+
+from torch_topological.utils import wrap_if_not_iterable
+
+
+class MultiScaleKernel(torch.nn.Module):
+    # TODO: more detailed description
+    r"""Implement the multi-scale kernel between two persistence diagrams
+
+    This class implements the multi-scale kernel between two persistence
+    diagrams as defined by Reininghaus et al. as
+
+    .. math::
+        k_\sigma(F,G) = \frac{1}{8 \pi \sigma} \sum_{\substack{p \in F\\q \in G}} exp{-\frac{\|p-q\|^2}{8\sigma}} - exp{-\frac{\|p-\overline{q}\|^2}{8\sigma}}
+
+    where :math:`z=(z_1, z_2)` and :math:`\overline{z}=(z_2, z_1)`
+
+    Attributes
+    ----------
+    sigma : float
+        Scale parameter
+
+    Methods
+    -------
+    forward(X, Y)
+        Compute the kernel value between two persistence diagrams
+
+    References
+    ----------
+    .. [Reininghaus15] J. Reininghaus, U. Bauer and R. Kwitt, "A Stable
+       Multi-Scale Kernel for Topological Machine Learning", *Proceedings
+       of the IEEE conference on computer vision and pattern recognition*,
+       pp. 4741--4748, 2015.
+    """
+    def __init__(self, sigma):
+        """Create a new MultiScaleKernel instance
+
+        Parameters
+        ----------
+        sigma : float
+            scale parameter of the kernel
+        """
+        super().__init__()
+
+        self.sigma = sigma
+
+    @staticmethod
+    def _check_upper(d):
+        """
+        Check if all points in the diagram are above the diagonal.
+        All points below the diagonal are 'swapped'.
+        """
+        is_upper = d[:, 0] < d[:, 1]
+        if not torch.all(is_upper):
+            d[~is_upper, 0] = d[~is_upper, 1]
+            d[~is_upper, 1] = d[~is_upper, 0]
+        return d
+
+    @staticmethod
+    def _mirror(x):
+        """Mirror one or multiple points of a persistence diagram at the
+        diagonal
+        """
+        if len(x.shape) > 1:
+            return x[:, [1, 0]]
+        # only a single point in the diagram
+        return x[[1, 0]]
+
+    @staticmethod
+    def _dist(x, y):
```

Comment:
Same comments as for other utility function apply; it would be perfectly sufficient to just have a multi-line comment here. Only user-exposed functions that are relevant for the API need to be searchable in Sphinx directly.

Reply:
Same as for the other utility functions